From f8e2ae246de1aec5dfdb100268470568bbfd44b2 Mon Sep 17 00:00:00 2001
From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Fri, 5 Sep 2014 11:03:50 -0400
Subject: [ethernet] sfc: Use canonical pointer type for MAC address in efx_set_mac_address()

Message-id: <1409915048-26438-15-git-send-email-nikolay@redhat.com>
Patchwork-id: 91417
O-Subject: [RHEL7.1 PATCH BZ 1110888 14/32] sfc: Use canonical pointer type for MAC address in efx_set_mac_address()
Bugzilla: 1110888
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Ben Hutchings <bhutchings@solarflare.com>

Functions such as is_valid_ether_addr() expect u8 *, so use that
instead of char *.

Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Signed-off-by: Shradha Shah <sshah@solarflare.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit e0b3ae30a2a4d4303ea394e4c2edd8f215dbb13b)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/sfc/efx.c b/drivers/net/ethernet/sfc/efx.c
index 6f7a86f..535663b 100644
--- a/drivers/net/ethernet/sfc/efx.c
+++ b/drivers/net/ethernet/sfc/efx.c
@@ -2113,7 +2113,7 @@ static int efx_set_mac_address(struct net_device *net_dev, void *data)
 {
  struct efx_nic *efx = netdev_priv(net_dev);
  struct sockaddr *addr = data;
- char *new_addr = addr->sa_data;
+ u8 *new_addr = addr->sa_data;
 
  if (!is_valid_ether_addr(new_addr)) {
   netif_err(efx, drv, efx->net_dev,
-- 
1.7.1